src/main/java/com/bookstore/service/OrderServiceImpl.java

```diff
@@ -0,0 +1,124 @@
+package com.bookstore.service;
+
+import com.bookstore.dto.OrderDto;
+import com.bookstore.dto.OrderItemDto;
+import com.bookstore.dto.OrderRequest;
+import com.bookstore.entity.CartItem;
+import com.bookstore.entity.Order;
+import com.bookstore.entity.OrderItem;
+import com.bookstore.entity.ShoppingCart;
+import com.bookstore.entity.User;
+import com.bookstore.exception.EntityNotFoundException;
+import com.bookstore.mapper.OrderItemMapper;
+import com.bookstore.mapper.OrderMapper;
+import com.bookstore.repository.OrderRepository;
+import com.bookstore.repository.ShoppingCartRepository;
+import jakarta.transaction.Transactional;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+
+    @Transactional
+    @Override
+    public OrderDto createOrder(OrderRequest request, User user) {
+        ShoppingCart shoppingCart = findShoppingCartByUser(user);
+
+        Order order = buildOrder(request, user, shoppingCart);
+        orderRepository.save(order);
+
+        return orderMapper.toDto(order);
+    }
+
+    @Override
+    public List<OrderDto> findAll(User user, Pageable pageable) {
+        Page<Order> ordersPage = orderRepository.findByUser(user, pageable);
+        List<Order> orders = ordersPage.getContent();
+        return orderMapper.toDtoList(orders);
+    }
+
+    @Transactional
+    @Override
+    public OrderDto updateOrderStatus(Long id, Order.Status status) {
+        Order order = orderRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Order not found with id: " + id));
+        order.setStatus(Order.Status.valueOf(status.name()));
```

Comment:
```suggestion
        order.setStatus(status);
```
Why do you complicate it?